cla_backend/apps/core/tests/admin/test_session_security.py

```diff
@@ -0,0 +1,202 @@
+        self.assert_logged_out(expected_time_to_log_out + self.TOLERANCE)
+
+        end = datetime.datetime.now()
+        delta = end - start
+
+        self.assertGreaterEqual(delta.seconds, expected_time_to_log_out - self.TOLERANCE)
+
+    def test_session_ends_on_browser_close(self):
+        """Tests that the session is ended when the broswer is closed.
+        """
+        # Login to the system and then close the broswer
+        self.login()
+        self.browser.close()
+
+        # Confirm that on reopening the browser and admin page that the session is
+        # not restored.
+        self.start_test_broswer()
+        self.navigate_to_admin_page()
+        self.assert_logged_out(self.WAIT)
+
+    def check_session_warning_displayed(self):
+        """Checks that the session security modal goes from hidden to visible within the
+        required timeframe.
+        """
+        start = datetime.datetime.now()
+
+        self.assert_warning_is_hidden()
+        self.assert_warning_element_visible(self.WARN_AFTER_SECONDS + self.TOLERANCE)
+
+        end = datetime.datetime.now()
+        delta = end - start
+
+        self.assertGreaterEqual(delta.seconds, self.WARN_AFTER_SECONDS)
+        self.assertLessEqual(delta.seconds, self.EXPIRE_AFTER_SECONDS)
+
+    def setup_admin_user(self):
+        """Creates a test user for the admin portal.
+        """
+        ContentType.objects.clear_cache()
+
+        if not User.objects.filter(username=self.TEST_LOGIN_CREDS).exists():
+            test_user = User.objects.create(username=self.TEST_LOGIN_CREDS)
+            test_user.set_password(self.TEST_LOGIN_CREDS)
+            test_user.is_staff = True
+            test_user.save()
+
+    def login(self):
+        """Performs admin portal login and confirms the presence of the session cookie.
+        """
+        self.navigate_to_admin_page()
+        self.browser.find_element("id", "id_username").send_keys(self.TEST_LOGIN_CREDS)
+        self.browser.find_element("id", "id_password").send_keys(self.TEST_LOGIN_CREDS)
+        self.browser.find_element_by_xpath('//input[@value="Log in"]').click()
+        self.confirm_logged_in()
+
+    def navigate_to_admin_page(self):
+        """Navigates to the admin portal.
+        """
+        self.browser.get('%s%s' % (self.live_server_url, '/admin/'))
+
+    def perform_keyboard_action(self):
+        """Simulates a user pressing the space bar within the browser.
+        """
+        action = ActionChains(self.browser)
+        action.key_down(Keys.SPACE)
+        action.key_up(Keys.SPACE)
+        action.perform()
+
+    def confirm_logged_in(self):
+        """Confirms the broswers is logged into the admin portal by check for the logout button
+        and the session cookie.
+        """
+        try:
+            WebDriverWait(self.browser, self.WAIT).until(
+                expected_conditions.presence_of_element_located((By.LINK_TEXT, "Log out")))
+        except Exception:
```

Comment:
Do you know what type of exception is thrown? This is me coming from a Java side here, but I think it would be better to catch the correct Exception, instead of a blanket of any Exception, better to be explicit. 

Same goes for all try catches that use Exception

Reply:
Any yes, good shout! Now done :)